packages/did-client/src/did-client.ts

```diff
@@ -1,90 +1,308 @@
+        //         },
+        //         {
+        //           id: 'did:ethr:0x61:0x599b3912A63c98dC774eF3E60282fBdf14cda748?context=0x67890c45e3ad1ba47c69f266d6c49c589b9d70de837e318c78ff43c7f0b00003',
+        //           type: 'Ed25519VerificationKey2018',
+        //           controller: 'did:ethr:0x61:0x599b3912A63c98dC774eF3E60282fBdf14cda748',
+        //           publicKeyBase58: '2E4cfzc9Kf2nvScMZ2bJwGKPn19TJYvPE98D8RCguqL6'
+        //         }
+        //     ],
+        //     assertionMethod: [
+        //         'did:ethr:0x61:0x599b3912A63c98dC774eF3E60282fBdf14cda748#controller',
+        //         'did:ethr:0x61:0x599b3912A63c98dC774eF3E60282fBdf14cda748?context=0x678904eb5c3f53d8506e7085dfbb0ef333c5f7d0769bcaf4ca2dc0ca46d00001',
+        //         'did:ethr:0x61:0x599b3912A63c98dC774eF3E60282fBdf14cda748?context=0x67890c45e3ad1ba47c69f266d6c49c589b9d70de837e318c78ff43c7f0b00003'
+        //     ],
+        //     authentication: [
+        //         'did:ethr:0x61:0x599b3912A63c98dC774eF3E60282fBdf14cda748#controller',
+        //         'did:ethr:0x61:0x599b3912A63c98dC774eF3E60282fBdf14cda748?context=0x678904eb5c3f53d8506e7085dfbb0ef333c5f7d0769bcaf4ca2dc0ca46d00001'
+        //     ],
+        //     service: [
+        //         {
+        //             id: 'did:ethr:0x61:0x599b3912A63c98dC774eF3E60282fBdf14cda748?context=0x84e5fb4eb5c3f53d8506e7085dfbb0ef333c5f7d0769bcaf4ca2dc0ca4698fd4&type=message',
+        //             type: 'VeridaMessage',
+        //             serviceEndpoint: 'https://db.testnet.verida.io:5002'
+        //         },
+        //         {
+        //             id: 'did:ethr:0x61:0x599b3912A63c98dC774eF3E60282fBdf14cda748?context=0xcfbf4621af64386c92c0badd0aa3ae3877a6ea6e298dfa54aa6b1ebe00769b28&type=database',
+        //             type: 'VeridaDatabase',
+        //             serviceEndpoint: 'https://db.testnet.verida.io:5002'
+        //         }
+        //     ]
+        //   }
     }
 
+    public authenticate(privateKey: string) {
+        throw new Error('Method not implemented.')
+    }
+    
     public getDid(): string | undefined {
-        return this.did
+        return this.vdaDid.did
     }
-
+    
     public getPublicKey(): string {
-        return this.wallet!.publicKey
+        return this.veridaWallet.publicKey
     }
 
     /**
-     * Save a DID document
+     * Save DIDDocument to the chain
      * 
-     * @param document 
-     * @returns 
+     * @param document Updated DIDDocuent
+     * @returns true if success.
      */
-    public async save(document: DIDDocument): Promise<boolean> {
-        if (!this.privateKey) {
-            throw new Error("Unable to save DIDDocument. No private key.")
+    public async saveDocument(document: DIDDocument | undefined): Promise<boolean> {
+        if (!this.didDoc) {
+            return Promise.reject("No original document")
```

Comment:
It should be possible to save a new document.

If `this.didDoc` doesn't exist, create a new empty document.

Reply:
Created an empty document when original document is not existed.